### PR TITLE
fix: 2107: Add missing schema fields and tests for ext_key_usage_oids

### LINKED ISSUE #2107: [Enhancement]: Missing support for `ext_key_usage_oids`
### Terraform Core Version

1.6.4

### Terraform Vault Provider Version

3.23.0

### Vault Server Version

1.15.2

### Affected Resource(s)

_No response_

### Expected Behavior

The Vault provider's `vault_pki_secret_backend_role` should expose a field for `ext_key_usage_oids` in line with the OpenAPI specification. Providing OIDs for key usage should render the desired OID under the Extended Key Usage OIDs field of the role:

Screenshot 2023-12-10 at 21 01 33

### Actual Behavior

The provider does not contain a field in `vault/resource_pki_secret_backend_role.go` for `ext_key_usage_oids` so the provider plan will fail as no schema would include that attribute. 
Screenshot 2023-12-10 at 21 08 59

### Relevant Error/Panic Output Snippet

_No response_

### Terraform Configuration Files

```
terraform {
  required_providers {
    vault = {
      source = "hashicorp/vault"
      version = "3.23.0"
    }
  }
}

provider "vault" {
 ..<snip>..
}

resource "vault_mount" "pki" {
  path                      = "pki"
  type                      = "pki"
  default_lease_ttl_seconds = 3600
  max_lease_ttl_seconds     = 86400
}

resource "vault_pki_secret_backend_role" "role" {
  backend          = vault_mount.pki.path
  name             = "my_role"
  ttl              = 3600
  allow_ip_sans    = true
  key_type         = "rsa"
  key_bits         = 4096
  allowed_domains  = ["example.com", "my.domain"]
  allow_subdomains = true
  allow_any_name = true

  policy_identifier {
    oid = "1.3.6.1.4.1.7.8"
    notice= "I am a user Notice"
  }
  policy_identifier {
    oid = "1.3.6.1.4.1.44947.1.2.4"
    cps ="https://example.com"
  }

   #ext_key_usage = ["1.3.6.1.4.1.311.4"] <<-- Does not render in role view or certificate
  ext_key_usage_oids = ["1.3.6.1.4.1.311.4"] <<-- Failure here.
}
```

### Steps to Reproduce

1. Create a main.tf with the contents of the provided configuration file.
2. Run `terraform plan`
3. Observe a schema validation error:
Screenshot 2023-12-10 at 21 08 19

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

Yes

### PR DESCRIPTION
**Pull request by austin-laney**
### Description

Relates OR Closes #2107 

### Checklist
- [x] Added [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md) entry (only for user-facing changes)
- [x] Acceptance tests where run against all supported Vault Versions

### Output from acceptance testing:

```
$ terraform-provider-vault % VAULT_TOKEN="TEST" VAULT_ADDR='http://127.0.0.1:8200' make testacc TESTARGS='-run=TestPkiSecretBackendRole'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test -run=TestPkiSecretBackendRole -timeout 30m ./...
?       github.com/hashicorp/terraform-provider-vault   [no test files]
?       github.com/hashicorp/terraform-provider-vault/cmd/coverage      [no test files]
?       github.com/hashicorp/terraform-provider-vault/cmd/generate      [no test files]
?       github.com/hashicorp/terraform-provider-vault/helper    [no test files]
?       github.com/hashicorp/terraform-provider-vault/internal/consts   [no test files]
?       github.com/hashicorp/terraform-provider-vault/internal/identity/group   [no test files]
?       github.com/hashicorp/terraform-provider-vault/internal/identity/mfa     [no test files]
?       github.com/hashicorp/terraform-provider-vault/internal/pki      [no test files]
ok      github.com/hashicorp/terraform-provider-vault/codegen   (cached) [no tests to run]
ok      github.com/hashicorp/terraform-provider-vault/internal/identity/entity  (cached) [no tests to run]
?       github.com/hashicorp/terraform-provider-vault/schema    [no test files]
ok      github.com/hashicorp/terraform-provider-vault/internal/provider (cached) [no tests to run]
ok      github.com/hashicorp/terraform-provider-vault/testutil  (cached) [no tests to run]
ok      github.com/hashicorp/terraform-provider-vault/util      (cached) [no tests to run]
ok      github.com/hashicorp/terraform-provider-vault/vault     16.497s
```

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by austin-laney** [comment]
Configuration including new `ext_key_usage_oids` attribute:
Screenshot 2023-12-10 at 21 42 13

Proper configuration of role:
Screenshot 2023-12-10 at 21 43 35

Added to list of extended key usage in generated certificate:
Screenshot 2023-12-10 at 21 44 56

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ## Unreleased
 
 BUGS:
+* fix `vault_pki_secret_backend_role` missing `ext_key_usage_oids` in schema ([#2107](https://github.com/hashicorp/terraform-provider-vault/pull/2108))_
```
Comment (fairclothjm) [inline CHANGELOG.md:4]: Can we move this to FEATURES please?

```suggestion
FEATURES:
* Add support for `ext_key_usage_oids` in `vault_pki_secret_backend_role` ([#2108](https://github.com/hashicorp/terraform-provider-vault/pull/2108))

BUGS:
```

File: vault/resource_pki_secret_backend_role_test.go
Comment (fairclothjm) [inline vault/resource_pki_secret_backend_role_test.go:1]: Getting a format error

```
gofmt needs running on the following files:
./vault/resource_pki_secret_backend_role_test.go
You can use the command: `make fmt` to reformat code.
make[1]: *** [fmtcheck] Error 1
make: *** [testacc-ent] Error 2
```

**Review by fairclothjm — CHANGES_REQUESTED** [review]
@austin-laney Thanks for the contribution! Could you please make the requested changes?

File: CHANGELOG.md (hunk 1 above)
Comment (austin-laney) [reply CHANGELOG.md]: Updated it most recent, commit. Thanks!

File: vault/resource_pki_secret_backend_role_test.go
Comment (austin-laney) [reply vault/resource_pki_secret_backend_role_test.go:1]: Ran `make fmt` and added changes.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,9 @@
 BUGS:
 * fix `vault_kv_secret_v2` drift when "data" is in secret name/path ([#2104](https://github.com/hashicorp/terraform-provider-vault/pull/2104))
 
+FEATURES:
+* Add support for `ext_key_usage_oids` in `vault_pki_secret_backend_role` ([#2108](https://github.com/hashicorp/terraform-provider-vault/pull/2108))
+
```
Comment (fairclothjm) [inline CHANGELOG.md:8]: nit: put features above bugs

```suggestion
FEATURES:
* Add support for `ext_key_usage_oids` in `vault_pki_secret_backend_role` ([#2108](https://github.com/hashicorp/terraform-provider-vault/pull/2108))

BUGS:
* fix `vault_kv_secret_v2` drift when "data" is in secret name/path ([#2104](https://github.com/hashicorp/terraform-provider-vault/pull/2104))

```

**Review by fairclothjm — APPROVED** [review]
LGTM! Thanks @austin-laney !

**Pull request merged**